controllers/platform/services.go

```diff
@@ -58,30 +55,32 @@ func (action *serviceAction) Handle(ctx context.Context, platform *operatorapi.S
 		return nil, err
 	}
 
-	if err := createDataIndexComponents(ctx, action.client, platform); err != nil {
-		return nil, err
+	if platform.Spec.Services.DataIndex != nil {
+		if err := createServiceComponents(ctx, action.client, platform, common.NewDataIndexService(platform)); err != nil {
+			return nil, err
+		}
+	}
+
+	if platform.Spec.Services.JobService != nil {
```

Comment:
same here, Should we get into this block if `platform.Spec.Services.JobService.Enabled == false`?